src/impl/tpls/KokkosBlas1_dot_tpl_spec_decl.hpp

```diff
@@ -47,14 +47,14 @@
 // Generic Host side BLAS (could be MKL or whatever)
 #ifdef KOKKOSKERNELS_ENABLE_TPL_BLAS
 
-extern "C" double               ddot_( const int* N, const double* x, const int* x_inc,
-                                       const double* y, const int* y_inc);
-extern "C" float                sdot_( const int* N, const float* x, const int* x_inc,
-                                       const float* y, const int* y_inc);
-extern "C" std::complex<double> zdotu_( const int* N, const std::complex<double>* x, const int* x_inc,
-                                       const std::complex<double>* y, const int* y_inc);
-extern "C" std::complex<float>  cdotu_( const int* N, const std::complex<float>* x, const int* x_inc,
-                                       const std::complex<float>* y, const int* y_inc);
+extern "C" double ddot_ ( const int* N, const double* x, const int* x_inc,
+                          const double* y, const int* y_inc);
+extern "C" float  sdot_ ( const int* N, const float* x, const int* x_inc,
+                          const float* y, const int* y_inc);
+extern "C" void   zdotu_( std::complex<double> *res, const int* N, const std::complex<double>* x, const int* x_inc,
```

Comment:
1. Does this way of returning a complex value work for all possible Fortran ABIs?
2. This interface still assumes that Fortran mangling uses a trailing underscore; this is not true on all platforms.
3. My objection about exposing extern declarations in user-facing header files still stands, though I think that would need to be addressed in a separate PR.

Reply:
1. I'm not convinced but I'm willing to keep an open mind.
2. Both TeuchosNumerics and my old TpetraCore GEMM and GEMV wrappers used Fortran mangling macros.  You could imitate either of them.
3. The BLAS implements 4 Scalar types; they are always the same.  Why not just wrap them in functions whose interface you declare in a particular namespace?  That's what I did with the TpetraCore GEMM and GEMV wrappers.